meson: replace hardcoded python command

Sigmanificient opened the pull request:
## Description of changes

- replace hardcoded python command

Fix:
- #313975 

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

arianvp committed:
nixos/wrappers: use normal mount for /run/wrappers

We want to get rid of specialFileSystems / earlyMountScript eventually and
there is no need to run this before systemd anymore now that
the wrappers themselves are set up in a systemd unit since https://github.com/NixOS/nixpkgs/pull/263203

Also this is needed to make soft-reboot work. We want to make sure
that we remount /run/wrappers with the nosuid bit removed on soft-reboot
but because @earlyMountScript@ happens in initrd, this wouldn't happen

Sigmanificient committed:
meson: replace hardcoded python command

Sigmanificient commented:
OH NO i didn't though it would rebuild `5000+` rebuilds

tie commented on pkgs/by-name/me/meson/package.nix:
```diff
@@ -118,7 +118,7 @@ python3.pkgs.buildPythonApplication rec {
   ])
   ++ [
-    ''HOME="$TMPDIR" python ./run_project_tests.py''
+    ''HOME="$TMPDIR" ${python3.interpreter} ./run_project_tests.py''
```
Shouldn’t this be using `python3.executable` instead to support cross-compilation?
I can’t really test this though because sadly pypy itself is broken :(

Sigmanificient replied:
> Shouldn’t this be using `python3.executable` instead to support cross-compilation? I can’t really test this
I dont know which would be better, `python3.interpreter` worked for me

Sigmanificient replied:
The package already has  `python3.interpreter` at some other places, we i think it would make sense to use it too here

![image](https://github.com/NixOS/nixpkgs/assets/53050011/d7ffc057-7a80-4b50-99aa-c8ac539ebd67)

tie replied:
Ah, that’s in the `checkPhase`, should be fine-ish then, unless someone takes the herculean effort of making `doCheck = true` by default in stdenv instead of always disabling it for build ≠ host case.
FWIW `python.interpreter` is an alias for `${python}/bin/${executable}`, so I *think* using `executable` would be better, assuming that `python` is already in PATH/nativeBuildInputs. Otherwise LGTM 👍

tie commented:
Meson tests probably also need a small hack since they expect python3 in PATH, unless that’s already in another PR or I’m missing something here—the changes in this PR are good, but I can’t build pypy310.pkgs.meson from your branch as-is, which I assume is your intention.

E.g.
<details>
<summary>diff pkgs/by-name/me/meson/package.nix</summary>

TODO: add pythonWrapper to outputChecks’s disallowedReferences?
```diff
diff --git a/pkgs/by-name/me/meson/package.nix b/pkgs/by-name/me/meson/package.nix
index b3cf409fa437..a3cb33f0c7fe 100644
--- a/pkgs/by-name/me/meson/package.nix
+++ b/pkgs/by-name/me/meson/package.nix
@@ -1,8 +1,8 @@
 { lib
 , stdenv
 , fetchFromGitHub
+, buildPackages
 , installShellFiles
-, coreutils
 , darwin
 , libxcrypt
 , openldap
@@ -15,6 +15,26 @@
 
 let
   inherit (darwin.apple_sdk.frameworks) AppKit Cocoa Foundation LDAP OpenAL OpenGL;
+
+  # Meson relies heavily on Python executable being named python3, so we give it
+  # a wrapper that executes appropriate executable.
+  pythonWrapper = buildPackages.writeCBin "python3" ''
+    #include <assert.h>
+    #include <errno.h>
+    #include <stdio.h>
+    #include <string.h>
+    #include <unistd.h>
+
+    const char* argv0 = "${python3.executable}";
+
+    int main(int argc, char *argv[]) {
+      assert(argc > 0);
+      argv[0] = argv0;
+      execvp(argv0, argv);
+      fprintf(stderr, "python3: %s: %s\n", argv0, strerror(errno));
+      return 1;
+    }
+  '';
 in
 python3.pkgs.buildPythonApplication rec {
   pname = "meson";
@@ -80,6 +100,8 @@ python3.pkgs.buildPythonApplication rec {
   nativeCheckInputs = [
     ninja
     pkg-config
+  ] ++ lib.optionals python3.isPyPy [
+    pythonWrapper
   ];
 
   checkInputs = [
@@ -98,11 +120,11 @@ python3.pkgs.buildPythonApplication rec {
   checkPhase = lib.concatStringsSep "\n" ([
     "runHook preCheck"
     ''
-      patchShebangs 'test cases'
+      patchShebangs --build 'test cases'
       substituteInPlace \
         'test cases/native/8 external program shebang parsing/script.int.in' \
         'test cases/common/273 customtarget exe for test/generate.py' \
-          --replace /usr/bin/env ${coreutils}/bin/env
+          --replace-fail /usr/bin/env ${buildPackages.coreutils}/bin/env
     ''
   ]
   # Remove problematic tests
@@ -139,7 +161,7 @@ python3.pkgs.buildPythonApplication rec {
     rm $out/nix-support/propagated-build-inputs
 
     substituteInPlace "$out/share/bash-completion/completions/meson" \
-      --replace "python3 -c " "${python3.interpreter} -c "
+      --replace-fail "python3 -c " "${python3.interpreter} -c "
   '';
 
   setupHook = ./setup-hook.sh;
```
</details>

alyssais committed:
linuxPackages_latest.rust-out-of-tree-module.updateScript: init

Added a version to the package to make the update script happy.

alyssais committed:
linuxPackages_latest.rust-out-of-tree-module: 0-unstable-2023-08-29 -> 0-unstable-2024-05-06

Updated for breaking changes in Linux 6.10.

paveloom committed:
python3Packages.undetected-chromedriver: init at 3.5.5

paveloom committed:
undetected-chromedriver: init at 126.0.6478.126

paveloom committed:
flaresolverr: init at 3.3.21

paveloom committed:
nixos/flaresolverr: initial commit

alyssais committed:
llvm: fix broken llvm-config-native for canExecute

Since a0b4b85bfaf0 ("llvm: Avoid cross compiling if the build platform
can execute host binaries"), the flags to get a working
llvm-config-native are not used when the build platform can execute
host binaries, resulting in a broken llvm-config-native, and therefore
a broken mesa, but since build can execute host, we don't need a
separate llvm-config-native at all — we can just use the normal
llvm-config.

Fixes: a0b4b85bfaf0 ("llvm: Avoid cross compiling if the build platform can execute host binaries")

pbsds committed:
flexget: 3.11.39 -> 3.11.40

Diff: https://github.com/Flexget/Flexget/compare/refs/tags/v3.11.39...v3.11.40

Changelog: https://github.com/Flexget/Flexget/releases/tag/v3.11.40

x123 committed:
phraze: Add updateScript and version test

eclairevoyant committed:
obs-studio: 30.1.2 -> 30.2.0

fabaff committed:
python312Packages.pytedee-async: 0.2.17 -> 0.2.20

Diff: https://github.com/zweckj/pytedee_async/compare/refs/tags/v0.2.17...v0.2.20

Changelog: https://github.com/zweckj/pytedee_async/releases/tag/v0.2.20

r-ryantm committed:
python312Packages.gpytorch: 1.11 -> 1.12

alyssais committed:
virtiofsd: 1.11.0 -> 1.11.1

K900 committed:
chromedriver: use autoPatchelfHook, clean up deps

K900 committed:
chromedriver: build from source where possible

r-ryantm committed:
minio: 2024-07-04T14-25-45Z -> 2024-07-16T23-46-41Z

4evy committed:
koboldcpp: backport old interfaces

Refs: https://github.com/NixOS/nixpkgs/pull/327102#discussion_r1677136308
Refs: https://github.com/NixOS/nixpkgs/pull/327102#discussion_r1677136419
Refs: https://github.com/NixOS/nixpkgs/pull/327102#discussion_r1677137201

Co-authored-by: SomeoneSerge <sergei.kozlukov@aalto.fi>

r-ryantm committed:
minify: 2.20.35 -> 2.20.37

herbetom committed:
yanic: set version variable during build

Setting this allows `yanic version` to report the package version.

4evy committed:
koboldcpp: fix x86_64-darwin build

Refs: https://github.com/NixOS/nixpkgs/pull/327102/checks?check_run_id=27511352788

4evy committed:
koboldcpp: rename interface arches -> cudaArches

Refs: https://github.com/NixOS/nixpkgs/pull/325540#discussion_r1677129990

Co-authored-by: SomeoneSerge <sergei.kozlukov@aalto.fi>

r-ryantm committed:
nanovna-saver: 0.6.3 -> 0.6.4

4evy committed:
koboldcpp: lib.optionalString -> lib.optionals (cublasSupport)

Refs: https://github.com/NixOS/nixpkgs/pull/325540#discussion_r1677129629

Co-authored-by: SomeoneSerge <sergei.kozlukov@aalto.fi>

4evy committed:
koboldcpp: makeWrapperArgs -> libraryPathWrapperArgs

`makeWrapperArgs` was quite ambiguous and unclear. `libraryPathWrapperArgs` is clearer on what it's supposed to actually do

pbsds committed:
flexget: downgrade to python311

pbsds committed:
flexget: 3.11.40 -> 3.11.41

Changelog: https://github.com/Flexget/Flexget/releases/tag/v3.11.41

pbsds committed:
flexget: migrate to by-name

adamcstephens committed:
maintainers: fix adamcstephens matrix

r-ryantm committed:
graphite-gtk-theme: 2024-04-28 -> 2024-07-15

adamcstephens committed:
CODEOWNERS: remove migrated lxc path

adamcstephens committed:
just: remove adamcstephens as maintainer

adamcstephens committed:
zfs: move maintainer adamcstephens to 2.2 only

adamcstephens committed:
consul: 1.19.0 -> 1.19.1

Aleksanaa committed:
alpaca: init at 0.9.6.1

OPNA2608 committed:
palemoon-bin: 33.2.0 -> 33.2.1

onny committed:
stalwart-mail: disable tests on aarch64

Ma27 committed:
mautrix-whatsapp: 0.10.7 -> 0.10.9

ChangeLogs:
* https://github.com/mautrix/whatsapp/releases/tag/v0.10.8
* https://github.com/mautrix/whatsapp/releases/tag/v0.10.9

Closes #320405

r-ryantm committed:
obs-studio-plugins.obs-vkcapture: 1.5.0 -> 1.5.1

r-ryantm committed:
gh-dash: 4.3.0 -> 4.4.0

dotlambda committed:
python312Packages.upb-lib: 0.5.7 -> 0.5.8

Diff: https://github.com/gwww/upb-lib/compare/refs/tags/0.5.7...0.5.8

Changelog: https://github.com/gwww/upb-lib/releases/tag/0.5.8

jian-lin committed:
emacsPackages.isearch-plus: trivialBuild -> melpaBuild

jian-lin committed:
emacsPackages.isearch-prop: trivialBuild -> melpaBuild

jian-lin committed:
emacsPackages.rect-mark: trivialBuild -> melpaBuild

rhelmot committed:
go: support FreeBSD

r-ryantm committed:
python312Packages.weaviate-client: 4.6.5 -> 4.6.7

konst-aa committed:
chickenPackages.chickenEggs: format

konst-aa committed:
chickenPackages.chickenEggs: fix allegro, dbus, math, and socket eggs

cole-h committed:
safeeyes: add missing setuptools dependency

Fixes:

    Traceback (most recent call last):
      File "/nix/store/km8nzjccd4r0g704is31q18qzl101g89-safeeyes-2.1.9/bin/.safeeyes-wrapped", line 6, in <module>
        from safeeyes.__main__ import main
      File "/nix/store/km8nzjccd4r0g704is31q18qzl101g89-safeeyes-2.1.9/lib/python3.12/site-packages/safeeyes/__main__.py", line 32, in <module>
        from safeeyes import utility
      File "/nix/store/km8nzjccd4r0g704is31q18qzl101g89-safeeyes-2.1.9/lib/python3.12/site-packages/safeeyes/utility.py", line 35, in <module>
        from distutils.version import LooseVersion
    ModuleNotFoundError: No module named 'distutils'

OPNA2608 committed:
nixos/lomiri: ubuntu_font_family -> ubuntu-classic

OPNA2608 committed:
tests/lomiri-system-settings: ubuntu_font_family -> ubuntu-classic

OPNA2608 committed:
tests/morph-browser: ubuntu_font_family -> ubuntu-classic

OPNA2608 committed:
tests/lomiri-filemanager-app: ubuntu_font_family -> ubuntu-classic

OPNA2608 committed:
tests/lomiri-calculator-app: ubuntu_font_family -> ubuntu-classic

OPNA2608 committed:
tests/lomiri-clock-app: ubuntu_font_family -> ubuntu-classic

OPNA2608 committed:
tests/teleports: ubuntu_font_family -> ubuntu-classic

mweinelt committed:
python312Packages.pyserial-asyncio-fast: 0.13 -> 0.14

https://github.com/home-assistant-libs/pyserial-asyncio-fast/releases/tag/0.14

r-ryantm committed:
wl-mirror: 0.16.4 -> 0.16.5

mweinelt committed:
python312Packages.holidays: 0.52 -> 0.53

https://github.com/vacanza/python-holidays/releases/tag/v0.53

pbsds committed:
Merge pull request #326474 from r-ryantm/auto-update/jmusicbot

jmusicbot: 0.4.1 -> 0.4.2

pbsds committed:
Merge pull request #326476 from r-ryantm/auto-update/gnmic

gnmic: 0.37.0 -> 0.38.0

fabaff committed:
python312Packages.python-kasa: 0.7.0.3 -> 0.7.0.5

Diff: https://github.com/python-kasa/python-kasa/compare/refs/tags/0.7.0.3...0.7.0.5

Changelog: https://github.com/python-kasa/python-kasa/blob/0.7.0.5/CHANGELOG.md

Co-Authored-By: Martin Weinelt <hexa@darmstadt.ccc.de>

pbsds committed:
Merge pull request #326514 from r-ryantm/auto-update/juju

juju: 3.5.1 -> 3.5.2

pbsds committed:
Merge pull request #326619 from r-ryantm/auto-update/buildkit

buildkit: 0.14.1 -> 0.15.0

pbsds committed:
Merge pull request #326643 from r-ryantm/auto-update/stylelint

stylelint: 16.6.1 -> 16.7.0

pbsds committed:
Merge pull request #326709 from r-ryantm/auto-update/python312Packages.coinmetrics-api-client

python312Packages.coinmetrics-api-client: 2024.2.6.16 -> 2024.7.11.18

emilylange committed:
Merge pull request #328141 from networkException/update-chromium

chromedriver,{ungoogled-,}chromium: 126.0.6478.126 -> 126.0.6478.182

mweinelt committed:
Merge pull request #328488 from mweinelt/pyserial-asyncio-fast

python312Packages.pyserial-asyncio-fast: 0.13 -> 0.14

pbsds committed:
Merge pull request #326712 from r-ryantm/auto-update/python312Packages.generic

python312Packages.generic: 1.1.2 -> 1.1.3

pbsds committed:
Merge pull request #326713 from r-ryantm/auto-update/python312Packages.typeshed-client

python312Packages.typeshed-client: 2.5.1 -> 2.6.0

pbsds committed:
Merge pull request #326768 from r-ryantm/auto-update/python312Packages.irc

python312Packages.irc: 20.4.1 -> 20.4.3

pbsds committed:
Merge pull request #326769 from r-ryantm/auto-update/ecs-agent

ecs-agent: 1.84.0 -> 1.85.0

pbsds committed:
Merge pull request #326812 from r-ryantm/auto-update/python312Packages.sphinxcontrib-tikz

python312Packages.sphinxcontrib-tikz: 0.4.19 -> 0.4.20

pbsds committed:
Merge pull request #326822 from r-ryantm/auto-update/python312Packages.pyvmomi

python312Packages.pyvmomi: 8.0.2.0.1 -> 8.0.3.0.1

pbsds committed:
Merge pull request #326850 from r-ryantm/auto-update/nest-cli

nest-cli: 10.4.0 -> 10.4.2

pbsds committed:
Merge pull request #326952 from r-ryantm/auto-update/fheroes2

fheroes2: 1.1.0 -> 1.1.1

risicle committed:
Merge pull request #327968 from alyssais/qemu-9.0.2

qemu: 9.0.1 -> 9.0.2

pbsds committed:
Merge pull request #326973 from r-ryantm/auto-update/libpointmatcher

libpointmatcher: 1.4.2 -> 1.4.3

pbsds committed:
Merge pull request #326975 from r-ryantm/auto-update/legit-web

legit-web: 0.2.2 -> 0.2.3

pbsds committed:
Merge pull request #326991 from r-ryantm/auto-update/python312Packages.whatthepatch

python312Packages.whatthepatch: 1.0.5 -> 1.0.6

pbsds committed:
Merge pull request #326997 from r-ryantm/auto-update/yaws

yaws: 2.1.1 -> 2.2.0

pbsds committed:
Merge pull request #327002 from r-ryantm/auto-update/igir

igir: 2.9.0 -> 2.9.2

pbsds committed:
Merge pull request #327017 from r-ryantm/auto-update/python312Packages.django-ninja

python312Packages.django-ninja: 1.1.0 -> 1.2.1

pbsds committed:
Merge pull request #327024 from r-ryantm/auto-update/python312Packages.packageurl-python

python312Packages.packageurl-python: 0.15.1 -> 0.15.3

pbsds committed:
Merge pull request #327199 from pyrox0/denose/adafruit-nrfutil

adafruit-nrfutil: remove nose and modernize

jopejoe1 committed:
Merge pull request #328487 from OPNA2608/fix/lomiri-ubuntu_font_family

Lomiri module & tests: ubuntu_font_family -> ubuntu-classic

r-ryantm committed:
cargo-make: 0.37.13 -> 0.37.14

nixpkgs-merge-bot[bot] committed:
Merge pull request #328334 from r-ryantm/auto-update/maa-assistant-arknights

maa-assistant-arknights: 5.3.1 -> 5.4.1

abysssol committed:
Merge pull request #328315 from redyf/update-ollama

ollama: 0.2.5 -> 0.2.7

uncenter committed:
vscode-extensions.nefrob.vscode-just-syntax: 0.3.0 -> 0.5.1

fabaff committed:
Merge pull request #327125 from fabaff/google-generativeai-bump

python312Packages.google-generativeai: 0.7.1 -> 0.7.2

fabaff committed:
Merge pull request #327123 from fabaff/google-cloud-workflows-bump

python312Packages.google-cloud-workflows: 1.14.3 -> 1.14.4

Mic92 committed:
Merge pull request #327815 from rhelmot/fix/freebsd

freebsd.mkDerivation: re-add missing hunk

fabaff committed:
Merge pull request #328354 from r-ryantm/auto-update/quark-engine

quark-engine: 24.6.1 -> 24.7.1

fabaff committed:
Merge pull request #328347 from r-ryantm/auto-update/malwoverview

malwoverview: 5.4.5 -> 6.0.0

fabaff committed:
Merge pull request #328313 from r-ryantm/auto-update/python312Packages.ollama

python312Packages.ollama: 0.2.1 -> 0.3.0

fabaff committed:
Merge pull request #328176 from r-ryantm/auto-update/trufflehog

trufflehog: 3.79.0 -> 3.80.0

fabaff committed:
Merge pull request #328168 from r-ryantm/auto-update/python312Packages.grpcio-testing

python312Packages.grpcio-testing: 1.64.1 -> 1.65.1

romildo committed:
Merge pull request #328431 from r-ryantm/auto-update/graphite-gtk-theme

graphite-gtk-theme: 2024-04-28 -> 2024-07-15

nixpkgs-merge-bot[bot] committed:
Merge pull request #328108 from r-ryantm/auto-update/awscli2

awscli2: 2.17.5 -> 2.17.13

fabaff committed:
Merge pull request #328131 from dotlambda/vunnel

vunnel: fix build

LeSuisse committed:
Merge pull request #328109 from r-ryantm/auto-update/brave

brave: 1.67.123 -> 1.67.134

Artturin committed:
Merge pull request #326324 from x123/x123-phraze-auto-update

phraze: set updateScript and add passthru.tests

r-ryantm committed:
files-cli: 2.13.85 -> 2.13.96

r-ryantm committed:
lbreakouthd: 1.1.8 -> 1.1.9

K900 committed:
kdePackages.kio: 6.4.0 -> 6.4.1

K900 committed:
kdePackages.kwidgetsaddons: 6.4.0 -> 6.4.1

K900 committed:
Merge pull request #328529 from K900/more-kde-backports

kdePackages: more small backports

K900 committed:
home-assistant-custom-lovelace-modules.android-tv-card: 3.8.1 -> 3.8.2

Diff: https://github.com/Nerwyn/android-tv-card/compare/3.8.1...3.8.2

K900 committed:
Merge pull request #328531 from K900/android-tv-card-3.8.2

home-assistant-custom-lovelace-modules.android-tv-card: 3.8.1 -> 3.8.2

adamcstephens committed:
Merge pull request #328430 from adamcstephens/acs/reduce

treewide: reduce adamcstephens maintenance

NickCao committed:
Merge pull request #328288 from r-ryantm/auto-update/kubeshark

kubeshark: 52.3.68 -> 52.3.69

NickCao committed:
Merge pull request #328287 from CobaltCause/update-cargo-llvm-cov

cargo-llvm-cov: 0.6.10 -> 0.6.11

NickCao committed:
Merge pull request #328338 from r-ryantm/auto-update/netclient

netclient: 0.24.2 -> 0.24.3

NickCao committed:
Merge pull request #328337 from r-ryantm/auto-update/netmaker

netmaker: 0.24.2 -> 0.24.3

NickCao committed:
Merge pull request #328336 from r-ryantm/auto-update/mpvScripts.videoclip

mpvScripts.videoclip: 0-unstable-2024-05-26 -> 0-unstable-2024-07-18

NickCao committed:
Merge pull request #328323 from r-ryantm/auto-update/c2patool

c2patool: 0.9.4 -> 0.9.5

NickCao committed:
Merge pull request #328322 from r-ryantm/auto-update/lact

lact: 0.5.4 -> 0.5.5

NickCao committed:
Merge pull request #328320 from r-ryantm/auto-update/hubble

hubble: 0.13.5 -> 0.13.6

NickCao committed:
Merge pull request #328317 from r-ryantm/auto-update/kubeseal

kubeseal: 0.27.0 -> 0.27.1

NickCao committed:
Merge pull request #328316 from r-ryantm/auto-update/sqlcmd

sqlcmd: 1.7.0 -> 1.8.0

NickCao committed:
Merge pull request #328303 from r-ryantm/auto-update/linuxhw-edid-fetcher

linuxhw-edid-fetcher: unstable-2023-05-08 -> 0-unstable-2023-05-08

NickCao committed:
Merge pull request #328299 from r-ryantm/auto-update/python312Packages.finvizfinance

python312Packages.finvizfinance: 1.0.0 -> 1.0.1

NickCao committed:
Merge pull request #328368 from r-ryantm/auto-update/python312Packages.qbittorrent-api

python312Packages.qbittorrent-api: 2024.5.63 -> 2024.7.64

NickCao committed:
Merge pull request #328367 from r-ryantm/auto-update/twitch-cli

twitch-cli: 1.1.23 -> 1.1.24

NickCao committed:
Merge pull request #328366 from r-ryantm/auto-update/ncspot

ncspot: 1.1.1 -> 1.1.2

NickCao committed:
Merge pull request #328358 from r-ryantm/auto-update/mill

mill: 0.11.8 -> 0.11.9

NickCao committed:
Merge pull request #328357 from r-ryantm/auto-update/python312Packages.dbt-redshift

python312Packages.dbt-redshift: 1.8.0 -> 1.8.1

NickCao committed:
Merge pull request #328344 from r-ryantm/auto-update/spicetify-cli

spicetify-cli: 2.36.14 -> 2.36.15

NickCao committed:
Merge pull request #328343 from r-ryantm/auto-update/reviewdog

reviewdog: 0.20.0 -> 0.20.1

NickCao committed:
Merge pull request #328407 from r-ryantm/auto-update/nanovna-saver

nanovna-saver: 0.6.3 -> 0.6.4

NickCao committed:
Merge pull request #328402 from r-ryantm/auto-update/minify

minify: 2.20.35 -> 2.20.37

NickCao committed:
Merge pull request #328384 from r-ryantm/auto-update/deck

deck: 1.39.2 -> 1.39.3

NickCao committed:
Merge pull request #328383 from r-ryantm/auto-update/nile

nile: 1.0.3-unstable-2024-06-08 -> 1.1.0-unstable-2024-07-15

NickCao committed:
Merge pull request #328376 from r-ryantm/auto-update/python312Packages.python-docs-theme

python312Packages.python-docs-theme: 2024.4 -> 2024.6

r-ryantm committed:
nh: 3.5.18 -> 3.5.19

NickCao committed:
Merge pull request #328460 from r-ryantm/auto-update/gh-dash

gh-dash: 4.3.0 -> 4.4.0

NickCao committed:
Merge pull request #328458 from r-ryantm/auto-update/obs-studio-plugins.obs-vkcapture

obs-studio-plugins.obs-vkcapture: 1.5.0 -> 1.5.1

NickCao committed:
Merge pull request #328493 from r-ryantm/auto-update/wl-mirror

wl-mirror: 0.16.4 -> 0.16.5

NickCao committed:
Merge pull request #328523 from r-ryantm/auto-update/files-cli

files-cli: 2.13.85 -> 2.13.96

LeSuisse committed:
Merge pull request #328440 from OPNA2608/update/palemoon-bin

palemoon-bin: 33.2.0 -> 33.2.1

emilytrau committed:
Merge pull request #326221 from emilytrau/cewler

cewler: init at 1.2.0

r-ryantm committed:
steampipe: 0.23.2 -> 0.23.3

nixpkgs-merge-bot[bot] committed:
Merge pull request #328528 from r-ryantm/auto-update/lbreakouthd

lbreakouthd: 1.1.8 -> 1.1.9

r-ryantm committed:
numix-icon-theme-circle: 24.04.22 -> 24.07.19

risicle committed:
Merge pull request #327365 from alyssais/mysql-8.0.38

mysql80: 8.0.37 -> 8.0.38

adamcstephens committed:
Merge pull request #328349 from mfrw/mfrw/just-1.32.0

just: 1.31.0 -> 1.32.0

wegank committed:
Merge pull request #328009 from konst-aa/chicken-5.4

chicken: 5.3.0 -> 5.4.0

r-ryantm committed:
android-studio: 2024.1.1.11 -> 2024.1.1.12

r-ryantm committed:
androidStudioPackages.canary: 2024.1.2.8 -> 2024.1.3.1

happysalada committed:
Merge pull request #328480 from r-ryantm/auto-update/python312Packages.weaviate-client

python312Packages.weaviate-client: 4.6.5 -> 4.6.7

happysalada committed:
Merge pull request #328450 from onny/stalwart-tests2

stalwart-mail: disable tests on aarch64

happysalada committed:
Merge pull request #328377 from r-ryantm/auto-update/tailwindcss-language-server

tailwindcss-language-server: 0.0.20 -> 0.0.21

pbsds committed:
Merge pull request #327867 from mjoerg/quickemu-4.9.6

quickemu: 4.9.5 -> 4.9.6

pbsds committed:
Merge pull request #327744 from malob/update-openai

python311Packages.openai: 1.35.13 -> 1.35.14

jian-lin committed:
Merge pull request #328470 from linj-fork/pr/melpaBuild-rect-mark

emacsPackages.rect-mark: trivialBuild -> melpaBuild

jian-lin committed:
Merge pull request #328468 from linj-fork/pr/clean-isearch-prop

emacsPackages.isearch-prop: trivialBuild -> melpaBuild

jian-lin committed:
Merge pull request #328466 from linj-fork/pr/melpaBuild-isearch-plus

emacsPackages.isearch-plus: trivialBuild -> melpaBuild

pbsds committed:
Merge pull request #327432 from getchoo/pkgs/rav1e/updateScript

rav1e: add updateScript

pbsds committed:
Merge pull request #327490 from Luflosi/update/linuxPackages.apfs

linuxPackages.apfs: 0.3.9 -> 0.3.10

pbsds committed:
Merge pull request #327585 from jtbx-patches/aerc-0.18.1

aerc: 0.18.0 -> 0.18.1

pbsds committed:
Merge pull request #327731 from niknetniko/gramps

gramps: 5.2.2 -> 5.2.3

pbsds committed:
Merge pull request #327730 from bbenno/graylog-5_2/update

graylog-5_2: 5.2.7 -> 5.2.9

fpletz committed:
ntpd-rs: 1.2.0 -> 1.2.2

fpletz committed:
ntpd-rs: disable testsuite, too flaky

r-ryantm committed:
obs-studio-plugins.obs-move-transition: 3.0.1 -> 3.0.2

r-ryantm committed:
cinnamon.pix: 3.4.1 -> 3.4.2

Also enable optional brasero / colord support.

https://github.com/linuxmint/pix/compare/3.4.1...3.4.2

r-ryantm committed:
xdg-desktop-portal-xapp: 1.0.7 -> 1.0.8

mweinelt committed:
home-assistant: 2024.7.2 -> 2024.7.3

https://github.com/home-assistant/core/releases/tag/2024.7.3

mweinelt committed:
python312Packages.homeassistant-stubs: 2024.7.2 -> 2024.7.3

https://github.com/KapJI/homeassistant-stubs/releases/tag/2024.7.3

r-ryantm committed:
cinnamon.cinnamon-session: 6.2.0 -> 6.2.1

r-ryantm committed:
python312Packages.yfinance: 0.2.40 -> 0.2.41

nixpkgs-merge-bot[bot] committed:
Merge pull request #328510 from r-ryantm/auto-update/cargo-make

cargo-make: 0.37.13 -> 0.37.14

reckenrode committed:
bustle: use gettext from nixpkgs

The vendored gettext fails to build with clang 16 on Darwin. The gettext in nixpkgs works.

jian-lin committed:
emacsPackages.ott-mode: trivialBuild -> melpaBuild

Also fix homepage and license.

reckenrode committed:
open-scq30: add CoreBluetooth framework on Darwin

reckenrode committed:
gnome-font-viewer: fix build with clang 16

r-ryantm committed:
python312Packages.distributed: 2024.7.0 -> 2024.7.1

r-ryantm committed:
vscodium: 1.90.2.24171 -> 1.91.1.24193

r-ryantm committed:
python312Packages.rotary-embedding-torch: 0.6.2 -> 0.6.4

marcusramberg committed:
Merge pull request #327467 from t4ccer/t4/non/fix-waf

marcusramberg committed:
Merge pull request #326751 from r-ryantm/auto-update/vault

marcusramberg committed:
Merge pull request #326722 from eclairevoyant/obs-update

drupol committed:
Merge pull request #327964 from atorres1985-contrib/bash-is-orphan

bash-5: remove dtzWill as maintainer [orphan]

drupol committed:
Merge pull request #328536 from r-ryantm/auto-update/nh

nh: 3.5.18 -> 3.5.19

afh committed:
doc: Remove indefinite article and ending period from example meta.description

so that meta.description examples shown in the documentation
align with recommendations given in the "Meta attributes" section
in pkgs/README.md.

The changes were made with the following commands:
nix run nixpkgs#silver-searcher -- -l0 'description\s*=\s*"([Aa]n?|[Tt]he)\s' doc \
  | xargs -0 nix run nixpkgs#gnused -- -i '' -Ee '/description/s/"([Aa]n?|[Tt]he)\s(.)/"\U\2/'
nix run nixpkgs#silver-searcher -- -l0 'description\s*=\s*".*\."' doc \
  | xargs -0 nix run nixpkgs#gnused -- -i '' -Ee '/description/s/\."/"/'

fabaff committed:
Merge pull request #328038 from r-ryantm/auto-update/termscp

termscp: 0.13.0 -> 0.14.0

Lassulus committed:
Merge pull request #327441 from getchoo/pkgs/ntpd-rs/by-name

ntpd-rs: migrate to by-name & add updateScript, 1.2.0 -> 1.2.2

reckenrode committed:
Merge pull request #328600 from reckenrode/open-scq30-fixes

open-scq30: add CoreBluetooth framework on Darwin

dotlambda committed:
Merge pull request #328590 from r-ryantm/auto-update/python312Packages.rotary-embedding-torch

python312Packages.rotary-embedding-torch: 0.6.2 -> 0.6.4

dotlambda committed:
Merge pull request #328585 from r-ryantm/auto-update/python312Packages.distributed

python312Packages.distributed: 2024.7.0 -> 2024.7.1

dotlambda committed:
Merge pull request #328575 from r-ryantm/auto-update/python312Packages.yfinance

python312Packages.yfinance: 0.2.40 -> 0.2.41

fpletz committed:
Merge pull request #328265 from trofi/egl-wayland-update

K900 committed:
Merge pull request #324894 from paveloom/flaresolverr

flaresolverr: init at 3.3.21 + nixos/flaresolverr: initial commit

r-ryantm committed:
androidStudioPackages.beta: 2024.1.1.10 -> 2024.1.2.9

nixpkgs-merge-bot[bot] committed:
Merge pull request #328541 from r-ryantm/auto-update/steampipe

steampipe: 0.23.2 -> 0.23.3

JohnRTitor committed:
Merge pull request #328548 from r-ryantm/auto-update/android-studio

android-studio: 2024.1.1.11 -> 2024.1.1.12

JohnRTitor committed:
Merge pull request #328549 from r-ryantm/auto-update/androidStudioPackages.canary

androidStudioPackages.canary: 2024.1.2.8 -> 2024.1.3.1

JohnRTitor committed:
Merge pull request #328613 from r-ryantm/auto-update/androidStudioPackages.beta

androidStudioPackages.beta: 2024.1.1.10 -> 2024.1.2.9

thiagokokada committed:
Merge pull request #327972 from atorres1985-contrib/remove-dtzwill

Treewide: Remove dtzwill

SuperSandro2000 committed:
Merge pull request #325827 from linsui/pjsip

pjsip: fix build on python 3.12

r-ryantm committed:
erigon: 2.60.2 -> 2.60.4

benhiemer committed:
vscode-extensions.42crunch.vscode-openapi: 4.25.3 -> 4.27.0

- Changelog: https://github.com/42Crunch/vscode-openapi/blob/master/CHANGELOG.md#version-4270-june-27-2024
- Comparing changes: https://github.com/42Crunch/vscode-openapi/compare/v4.25.3...v4.27.0

gepbird committed:
osu-lazer-bin: 2024.718.0 -> 2024.718.1

Ma27 committed:
Merge pull request #327806 from sumnerevans/matrix-synapse-1.111

matrix-synapse-unwrapped: 1.110.0 -> 1.111.0

K900 committed:
Merge pull request #326845 from Luflosi/init/ubootOrangePiZero3

ubootOrangePiZero3: init

gepbird committed:
osu-lazer: 2024.718.0 -> 2024.718.1

Aleksanaa committed:
Merge pull request #328436 from Aleksanaa/alpaca

alpaca: init at 0.9.6.1

romildo committed:
Merge pull request #328543 from r-ryantm/auto-update/numix-icon-theme-circle

numix-icon-theme-circle: 24.04.22 -> 24.07.19

thiagokokada committed:
Merge pull request #328639 from gepbird/osu-lazer-2024.718.1

osu-lazer{,-bin}: 2024.718.0 -> 2024.718.1

mweinelt committed:
Merge pull request #328608 from afh/doc-description-examples

doc: Remove indefinite article and ending period from example meta.description

symphorien committed:
Merge pull request #327214 from pyrox0/unbreak-paperwork

paperwork: unbreak

r-ryantm committed:
silice: 0-unstable-2024-06-23 -> 0-unstable-2024-07-15

SomeoneSerge committed:
cudaPackages.writeGpuTestPython: allow a selector for `libraries` to accommodate different python versions

SomeoneSerge committed:
cudaPackages.writeGpuTestPython: sync the attr and the filesystem paths

SomeoneSerge committed:
python311Packages.torch.tests.tester-*Available: unbreak for non-default python package sets

SomeoneSerge committed:
python3Packages.torch.tests.*compile*: init

superherointj committed:
Merge pull request #328636 from benhiemer/bump-42crunch-vscode-openapi

vscode-extensions.42crunch.vscode-openapi: 4.25.3 -> 4.27.0

mweinelt committed:
Merge pull request #328570 from NixOS/home-assistant

home-assistant: 2024.7.2 -> 2024.7.3

superherointj committed:
Merge pull request #328361 from datafoo/vscode-extensions.streetsidesoftware.code-spell-checker

vscode-extensions.streetsidesoftware.code-spell-checker: 4.0.3 -> 4.0.4

superherointj committed:
Merge pull request #328511 from uncenter/pr-9bd75400

vscode-extensions.nefrob.vscode-just-syntax: 0.3.0 -> 0.5.1

SomeoneSerge committed:
cudaPackages.writeGpuTestPython: accept makeWrapperArgs

pbsds committed:
Merge pull request #328649 from r-ryantm/auto-update/silice

silice: 0-unstable-2024-06-23 -> 0-unstable-2024-07-15

pbsds committed:
Merge pull request #328120 from r-ryantm/auto-update/radicle-httpd

radicle-httpd: 0.13.0 -> 0.14.0

reckenrode committed:
Merge pull request #328588 from reckenrode/bustle-fixes

bustle: use gettext from nixpkgs

reckenrode committed:
Merge pull request #328592 from reckenrode/gnome-font-viewer-fixes

gnome-font-viewer: fix build with clang 16

yorickvP committed:
Merge pull request #328043 from r-ryantm/auto-update/victoriametrics

victoriametrics: 1.101.0 -> 1.102.0

pbsds committed:
Merge pull request #325938 from pbsds/bump-flexget-1720558165

flexget: 3.11.39 -> 3.11.41

leona-ya committed:
Merge pull request #328240 from balacij/master

vikunja: 0.24.0 -> 0.24.1

A1ca7raz committed:
applet-window-buttons6: init at 0.13.0

jian-lin committed:
Merge pull request #328581 from linj-fork/pr/melpaBuild-ott-mode

emacsPackages.ott-mode: trivialBuild -> melpaBuild

Lassulus committed:
Merge pull request #328406 from herbetom/yanic-add-version-var

yanic: set version variable during build

arianvp committed:
Merge pull request #328653 from arianvp/later-wrappers

nixos/wrappers: use normal mount for /run/wrappers

adamcstephens committed:
Merge pull request #328434 from adamcstephens/consul/1.19.1

consul: 1.19.0 -> 1.19.1

fpletz committed:
Merge pull request #320075 from illustris/pve-lxc-ping-fix

K900 committed:
Merge pull request #328324 from A1ca7raz/ab6

applet-window-buttons6: init at 0.13.0

pbsds committed:
Merge pull request #326950 from florensie/update-openrct2

openrct2: 0.4.11 -> 0.4.12

Ericson2314 committed:
Merge pull request #327817 from rhelmot/freebsd-out-of-tree

freebsd.mkDerivation: do not crash on packages not using FreeBSD src

bobby285271 committed:
Merge pull request #328566 from r-ryantm/auto-update/cinnamon.pix

cinnamon.pix: 3.4.1 -> 3.4.2

SomeoneSerge committed:
Merge pull request #328661 from SomeoneSerge/feat/torch-compile-test

python312Packages.torch: test torch.compile

fpletz committed:
Merge pull request #328392 from OPNA2608/powerpc64/libressl

fpletz committed:
Merge pull request #327207 from SFrijters/clever-tools-fix-cross

SomeoneSerge committed:
Merge pull request #328413 from DontEatOreo/interface-backport-koboldcpp

koboldcpp: interface backport and small fixes

r-ryantm committed:
iosevka: 30.3.2 -> 30.3.3

drupol committed:
Merge pull request #328701 from r-ryantm/auto-update/iosevka

iosevka: 30.3.2 -> 30.3.3

Lassulus committed:
Merge pull request #241373 from nikstur/qemu-vm-simplify-nix-store-image

nixos/qemu-vm: simplify building nix store image

happysalada committed:
Merge pull request #328625 from r-ryantm/auto-update/erigon

erigon: 2.60.2 -> 2.60.4

drupol committed:
Merge pull request #328565 from r-ryantm/auto-update/obs-studio-plugins.obs-move-transition

obs-studio-plugins.obs-move-transition: 3.0.1 -> 3.0.2

NickCao committed:
Merge pull request #328455 from Ma27/bump-mautrix-whatsapp

mautrix-whatsapp: 0.10.7 -> 0.10.9

Sigmanificient commented:
This pr
